### ISSUE TITLE
UploadCollectionItem: edit and delete button should have unique accessible names (A11y)

### ISSUE THREAD

**Issue by summer2200** [SAP/ui5-webcomponents-react#6906]
### Describe the bug

When `UploadCollection` is set to `Detail` type, each item in the collection has a edit button and a delete button in the end, and these buttons should have unique accessible names.
Currently the buttons have no accessible names.

### Isolated Example

https://stackblitz.com/edit/github-cpmz8rrr?file=src%2FApp.tsx

### Expected Behaviour

Edit button and delete button of each item in `UploadCollection` should have unique accessible names.

### Screenshots or Videos

<img width="1015" alt="Image" src="https://github.com/user-attachments/assets/892f8249-3c72-417a-b134-032419387040" />

### UI5 Web Components for React Version

2.5.0

### UI5 Web Components Version

2.5.1

### Browser

Chrome

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by Lukas742** [comment]
Thanks for reporting! I'll forward this issue to our UI5 Web Components Colleagues as the affected component is developed in their repository.
